patches/tModLoader/Terraria/ModLoader/IO/WorldIO.cs

```diff
@@ -291,6 +293,73 @@ internal static class WorldIO
 			}
 		}
 
+		internal static List<TagCompound> SaveModEventData() {
+			var list = new List<TagCompound>();
+
+			foreach (var modEvent in ModEventLoader.ModEvents) {
+				var data = modEvent.Save();
+
+				ModInvasion modInvasion = modEvent as ModInvasion;
+
+				list.Add(new TagCompound {
+					["mod"] = modEvent.Mod.Name,
+					["name"] = modEvent.Name,
+					["active"] = modEvent.Active,
+					["isInvasion"] = modInvasion != null,
+					["progress"] = modInvasion != null ? modInvasion.Progress : -1f,
+					["data"] = data
+				});
+			}
```

Comment:
was trying to follow suit as how other ModTypes save, felt like just adding 2 fields would be appropriate because ModInvasion is and will always be the only major subtype deriving from ModEvent - unless you can give me pointers on what exactly to do here when you say inheritance?